portfolio/src/main/webapp/index.html

```diff
@@ -10,6 +10,12 @@
   <body onload="getGreeting()">
     <div id="content">
       <div id="greeting-container"></div>
+      <form action="/data" method="POST">
```

Comment:
Same here. Form submitting is outdated. Use the service to provide data through http requests, and use client code to control the navigation.